x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/components/delete_modal.tsx

```diff
@@ -17,125 +17,116 @@ import {
+                values: { name: item?.name, message: error.message },
+              }
+            )
+          );
+          onCancel(true);
+        },
+        onSuccess: () => {
+          toasts.addSuccess(
+            i18n.translate(
+              'xpack.securitySolution.hostIsolationExceptions.deletionDialog.deleteSuccess',
+              {
+                defaultMessage:
+                  '"{name}" has been removed from the Host isolation exceptions list.',
+                values: { name: item?.name },
+              }
+            )
+          );
+          onCancel(true);
+        },
+      }
+    );
 
-  // show toast for failures
-  useEffect(() => {
-    if (deleteError) {
-      toasts.addDanger(
-        i18n.translate(
-          'xpack.securitySolution.hostIsolationExceptions.deletionDialog.deleteFailure',
-          {
-            defaultMessage:
-              'Unable to remove "{name}" from the Host isolation exceptions list. Reason: {message}',
-            values: { name: exception?.name, message: deleteError.message },
-          }
-        )
-      );
-    }
-  }, [deleteError, exception?.name, toasts]);
+    return (
+      <EuiModal onClose={() => onCancel()}>
+        <EuiModalHeader data-test-subj="hostIsolationExceptionsDeleteModalHeader">
+          <EuiModalHeaderTitle>
+            <FormattedMessage
+              id="xpack.securitySolution.hostIsolationExceptions.deletionDialog.title"
+              defaultMessage="Delete Host isolation exception"
+            />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
 
-  return (
-    <EuiModal onClose={onCancel}>
-      <EuiModalHeader data-test-subj="hostIsolationExceptionsDeleteModalHeader">
-        <EuiModalHeaderTitle>
-          <FormattedMessage
-            id="xpack.securitySolution.hostIsolationExceptions.deletionDialog.title"
-            defaultMessage="Delete Host isolation exception"
-          />
-        </EuiModalHeaderTitle>
-      </EuiModalHeader>
+        <EuiModalBody data-test-subj="hostIsolationExceptionsFilterDeleteModalBody">
+          <EuiText>
+            <p>
+              <FormattedMessage
+                id="xpack.securitySolution.hostIsolationExceptions.deletionDialog.subtitle"
+                defaultMessage='You are deleting exception "{name}".'
+                values={{ name: <b className="eui-textBreakWord">{item?.name}</b> }}
+              />
+            </p>
+            <p>
+              <FormattedMessage
+                id="xpack.securitySolution.hostIsolationExceptions.deletionDialog.confirmation"
+                defaultMessage="This action cannot be undone. Are you sure you wish to continue?"
+              />
+            </p>
+          </EuiText>
+        </EuiModalBody>
 
-      <EuiModalBody data-test-subj="hostIsolationExceptionsFilterDeleteModalBody">
-        <EuiText>
-          <p>
+        <EuiModalFooter>
+          <EuiButtonEmpty
+            onClick={() => onCancel()}
```

Comment:
YOu have a few callbacks that are not memoized with `useCallback` - here and below.

Reply:
see my previous response. This doesn't need useCallback.

Reply:
as long as you understand that it could be causing un-necessary component re-renders due to React's strict equality props checker.